pilot/pkg/serviceregistry/serviceentry/servicediscovery.go

```diff
@@ -330,6 +330,53 @@ func (s *ServiceEntryStore) serviceEntryHandler(old, curr config.Config, event m
 	for _, svc := range nonDNSServices {
 		keys[instancesKey{hostname: svc.Hostname, namespace: curr.Namespace}] = struct{}{}
 	}
+
+	di := map[instancesKey]map[configKey][]*model.ServiceInstance{}
```

Comment:
Wouldn't this create so much churn as indicated in maybeRefreshIndexes comment - any update will cause reconversion  multiple times? refreshIndexes will ensure it is done lazily?

Reply:
Incremental update of serviceentry is very necessary. anything needs to be updated are associated with the serviceentry. maybeRefreshIndexes takes too much time.